packages/react-components/react-virtualizer/src/hooks/useMeasureList.ts

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+
+export interface IndexedResizeCallbackElement {
+  handleResize: () => void;
+}
+/**
+ * Provides a way of automating size in the virtualizer
+ * Returns
+ * `width` - element width ref (0 by default),
+ * `height` - element height ref (0 by default),
+ * `measureElementRef` - a ref function to be passed as `ref` to the element you want to measure
+ */
+export function useMeasureList<
+  TElement extends HTMLElement & IndexedResizeCallbackElement = HTMLElement & IndexedResizeCallbackElement,
+>(currentIndex: number, refLength: number, totalLength: number, defaultItemSize: number) {
+  const widthArray = React.useRef(new Array(totalLength).fill(defaultItemSize));
+  const heightArray = React.useRef(new Array(totalLength).fill(defaultItemSize));
+
+  const refArray = React.useRef<Array<TElement | undefined | null>>([]);
+  const { targetDocument } = useFluent();
+
+  // the handler for resize observer
+  const handleIndexUpdate = React.useCallback(
+    (index: number) => {
+      const boundClientRect = refArray.current[index]?.getBoundingClientRect();
+      const containerWidth = boundClientRect?.width;
+      widthArray.current[currentIndex + index] = containerWidth || defaultItemSize;
+
+      const containerHeight = boundClientRect?.height;
+      heightArray.current[currentIndex + index] = containerHeight || defaultItemSize;
+    },
+    [currentIndex, defaultItemSize],
+  );
+
+  const handleResize = (entries: ResizeObserverEntry[]) => {
```

Comment:
Let's differentiate the two `handleResize` functions we have. One is the callback for RO and the other an injected function on our observed elements. Might help with clarity.